request-manager/request/manager.go

```diff
@@ -353,6 +394,29 @@ func (m *manager) Status(requestId string) (proto.RequestStatus, error) {
 	return reqStatus, nil
 }
 
+func (m *manager) Finish(requestId string, finishParams proto.FinishRequestParams) error {
+	req, err := m.Get(requestId)
+	if err != nil {
+		return err
+	}
+
+	now := time.Now()
+	req.FinishedAt = &now
+	prevState := req.State
+	req.State = finishParams.State
+
+	// This will only update the request if the current state is RUNNING.
+	err = m.updateRequest(req, proto.STATE_RUNNING)
+	if err != nil {
+		if prevState != proto.STATE_RUNNING {
```

Comment:
Would this ever happen here? Why?

Reply:
I looked into it a bit - as the code is now I don't think we can ever Finish a request when its state isn't running. I want to keep the check on the error, though, because if a bug's introduced later it's easier to debug `request not in __ state (state = __)` than `cannot find record to update`.